src/api/tokenList/TokenListApi.ts

```diff
@@ -13,21 +26,67 @@ export interface TokenList {
 export class TokenListApiImpl implements TokenList {
   public networkIds: number[]
   private _tokensByNetwork: { [networkId: number]: TokenDetails[] }
+  private _tokenAddressNetworkSet: Set<string>
 
   public constructor(networkIds: number[]) {
     this.networkIds = networkIds
 
     // Init the tokens by network
     this._tokensByNetwork = {}
+    this._tokenAddressNetworkSet = new Set<string>()
+
     networkIds.forEach(networkId => {
       // initial value
-      this._tokensByNetwork[networkId] = getTokensByNetwork(networkId)
+      const tokenList = getTokensByNetwork(networkId).concat(this.loadUserTokenList(networkId))
+      this._tokensByNetwork[networkId] = tokenList
+
+      tokenList.forEach(({ address }) => {
+        this._tokenAddressNetworkSet.add(
+          TokenListApiImpl.constructAddressNetworkKey({ tokenAddress: address, networkId }),
```

Comment:
i think because he's using a `lambda` function (`=>`) this will be applied to the function itself not the calling entity, static or not

Reply:
the lamba being inside the static method being the problem

Reply:
We are inside the constructor, `this` is class instance, lambdas have `this` from lexical (parent) scope -- same classs instance
I don't understand the problem